### PR TITLE
Add Credential Loss and Key Mobility section to security considerations

### LINKED ISSUE #1020: User agent / key management [authenticator] interoperability requirements?
One of the primary challenges with previous asymmetric cryptography browser based authentication schemes has been key management (and notably keys being locked into devices/applications/user agents).

#### Some of the Historic Issues:
- Keys lost on OS upgrade / re-installs / wipes / device loss.
- Keys not usable across form factors (mobile app vs desktop browser).
- Keys not usable across devices from different walled ecosystems (apple, google etc.).
- Keys not usable across user-agents / applications.
- Security of keys compromised by malware on devices (sure addressable if user is going to use an external USB key store but these are problematic on many form factors).

Has consideration been given to requiring that user agents support a minimum set of standard APIs for connecting to external software/hardware authenticators / key stores (such that key managers [authenticators] are made available to the user agent via a simple standardised minimum set of APIs / easy user registration).

Note, none of this requires sharing private key material between devices (this is simply about ensuring that 'user agents' can connect to compliant 'authenticators' in standardised ways rather than allowing user agent vendors to support this spec by simply extending their existing walled-ecosystem password/key-management solution).

### Network API
Require that user agents support a Network API which requests authentication (where supported by platform i.e. all network connected devices).
- e.g. Allow user to authorise website login on a desktop device by biometric authentication / presence confirmation / pin entry / etc. on a mobile device (obviously dependant on mobile device app push notifications etc.)

### Local Software API
Require that user agents allow registering of key manager locally. I.e. a conformant 3rdp party password manager (where supported by platform).
- e.g. Key manager could be a local password manager (used across browsers / apps) or software that defers authentication to a more trusted device (i.e. mobile / specialised device etc.) via either a network or local (physical / USB / Bluetooth / NFC) communication.

## Why

Unless the specification mandates and standardises the minimum level of support for external key management there is potentially a risk that these keys will be locked into single platforms like passwords are today (chrome vs safari vs msie password management solutions).

## References

[Ref: 2.2:](https://www.w3.org/TR/webauthn/#conforming-authenticators) As described in §1.1 Use Cases, an authenticator may be implemented in the operating system underlying the User Agent, or in external hardware, or a combination of both.

[Ref: 5.1.3:](https://www.w3.org/TR/webauthn/#createCredential) During the above process, the user agent SHOULD show some UI to the user to guide them in the process of selecting and authorizing an authenticator.

### PR DESCRIPTION
**Pull request by emlun**
This fixes #1020.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1057.html" title="Last updated on Sep 12, 2018, 6:18 AM GMT (2119fbc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1057/5bdcae4...2119fbc.html" title="Last updated on Sep 12, 2018, 6:18 AM GMT (2119fbc)">Diff</a>

### CODE REVIEW

**Comment by nadalin** [comment]
@equalsJeffH @selfissued Please review

File: index.bs [hunk 1]
```diff
@@ -5265,6 +5265,24 @@ permissions for certain extensions. An example is the location extension (see [[
 which SHOULD make use of the existing browser permissions framework for the Geolocation API.
 
 
+## Credential Loss and Key Mobility ## {#credential-loss-key-mobility}
+
+This specification defines no protocol for backing up [=credential private keys=], or for sharing them between [=authenticators=].
+In general, it is expected that a [=credential private key=] never leaves the [=authenticator=] that created it. Losing an
+[=authenticator=] therefore in general means losing all [=public key credential|credentials=] [=bound credential|bound=] to the
```
Comment (selfissued) [inline index.bs]: Put commas around "in general".

Comment (emlun) [reply index.bs]: Done!

File: index.bs [hunk 2]
```diff
@@ -5269,6 +5269,24 @@ permissions for certain extensions. An example is the location extension (see [[
 which SHOULD make use of the existing browser permissions framework for the Geolocation API.
 
 
+## Credential Loss and Key Mobility ## {#credential-loss-key-mobility}
+
+This specification defines no protocol for backing up [=credential private keys=], or for sharing them between [=authenticators=].
```
Comment (equalsJeffH) [inline index.bs]: Nit:
This specification does not define a means for backing up ...

**Comment by equalsJeffH** [comment]
On the 12-Sep-2018 call, the WG agreed to just merge this.

**Pull request merged**